src/main/kotlin/org/jetbrains/intellij/dependency/MavenPluginsRepository.kt

```diff
@@ -1,41 +1,55 @@
 package org.jetbrains.intellij.dependency
 
+import groovy.lang.Closure
 import groovy.transform.CompileStatic
 import org.gradle.api.Project
+import org.gradle.util.ConfigureUtil
 import org.jetbrains.intellij.debug
 import java.io.File
 import java.net.URI
 
 @CompileStatic
-class MavenPluginsRepository(private val repositoryUrl: String) : PluginsRepository {
+class MavenPluginsRepository(private val repositoryUrl: String?, private val mavenClosure: Closure<Any>? = null) :
```

Comment:
This kind of constructors is misleading, I'd split it into two separate classes or at least in to factory functions

Reply:
Done